File: common/src/main/java/com/ultreon/mods/advanceddebug/api/client/registry/IFormatterRegistry.java
package com.ultreon.mods.advanceddebug.api.client.registry;

import com.ultreon.mods.advanceddebug.api.client.menu.Formatter;
import org.jetbrains.annotations.Nullable;

public interface IFormatterRegistry {
    <T> Formatter<T> register(Formatter<T> formatter);

    @Nullable
    Formatter<?> identify(Class<?> aClass);
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/api/client/registry/IFormatterRegistry.kt
package com.ultreon.mods.advanceddebug.api.client.registry

import com.ultreon.mods.advanceddebug.api.client.menu.Formatter
import kotlin.reflect.KClass

interface IFormatterRegistry {
    fun <T : Any> register(formatter: Formatter<T>): Formatter<T>
    fun identify(`class`: Class<*>): Formatter<*>?
    fun identify(kClass: KClass<*>): Formatter<*>?
}